src/graph/ChcGraph.cc

```diff
@@ -410,11 +410,12 @@ ChcDirectedHyperGraph::VertexContractionResult ChcDirectedHyperGraph::contractVe
     std::transform(outgoingEdges.begin(), outgoingEdges.end(), std::back_inserter(result.outgoing), [this](EId eid) {
         return this->getEdge(eid);
     });
-    std::size_t incomingIndex = 0;
-    for (EId incomingId : incomingEdges) {
+    for (std::size_t incomingIndex = 0; incomingIndex < incomingEdges.size(); ++incomingIndex) {
```

Comment:
Why change the loop conditions?

Reply:
To make sure I don't forget to increment the index, which was the bug in the previous version of the code.